docker/owlbot/nodejs/Dockerfile

```diff
@@ -16,39 +16,92 @@
 
 # build from the root of this repo:
 # docker build -t gcr.io/repo-automation-bots/owlbot-nodejs -f docker/owlbot/nodejs/Dockerfile .
-FROM python:3.10.16-bookworm
+FROM marketplace.gcr.io/google/debian12:latest
 WORKDIR /
 
+## Install Build Essentials, ca-certificates, and Python 3.10.16
+
+# This step installs all necessary tools and libraries to compile Python from source.
+# It also includes 'ca-certificates' to ensure secure HTTPS connections for downloads.
+# Git is also included here as it's a common dependency.
 RUN apt-get update && \
-    apt-get upgrade -y
+    apt-get install -y --no-install-recommends \
+    ca-certificates \
+    build-essential \
+    zlib1g-dev \
+    libncurses5-dev \
+    libgdbm-dev \
+    libnss3-dev \
+    libssl-dev \
+    libreadline-dev \
+    libffi-dev \
+    libsqlite3-dev \
+    wget \
+    libbz2-dev \
+    tk-dev \
+    liblzma-dev \
+    curl \
+    git && \
+    rm -rf /var/lib/apt/lists/*
+
+# Define the Python version for easier updates and download/install Python from source.
```

Comment:
Why did you chose to install Python from source rather than with `apt-get`?

Reply:
Changed.

Reply:
Thanks!